uplink/ecclient/client.go

```diff
@@ -213,42 +213,43 @@ func (ec *ecClient) putPiece(ctx, parent context.Context, limit *pb.AddressedOrd
 		)
 		return nil, err
 	}
+
 	defer func() {
-		if ctx.Err() != nil || err != nil {
-			hash = nil
-			err = errs.Combine(err, upload.Cancel(ctx))
-			return
+		if err != nil {
+			errs.Combine(err, upload.Cancel(ctx))
```

Comment:
also return after cancel so we don't call upload.Commit when we cancel

Reply:
oh, yes, that's a mistake, which happens because of doing these things while in a meeting :sweat_smile: